include/nuttx/progmem.h

```diff
@@ -49,96 +49,119 @@ extern "C"
  * Public Functions Definitions
  ****************************************************************************/
 
+/* NOTE: Since vendors use the terms page, block, and sector interchangably,
+ * it causes confusion and therefore a neutral term is used. We use the term
+ * "index" to refer to the ordinal number of erasable units of memory.
+ */
+
 /****************************************************************************
- * Name: up_progmem_neraseblocks
+ * Name: up_progmem_maxeraseindex
  *
  * Description:
- *   Return number of erase blocks
+ *   Return the total number of erasable units of memory.
```

Comment:
Add the fail return value 0

Reply:
@davids5 Could this even fail? I suppose 0 means that there are no programmable units of memory?

Reply:
yep